yang/frr-nexthop.yang

```diff
@@ -100,110 +105,143 @@ module frr-nexthop {
    * Nexthop object
    */
 
-  grouping frr-nexthop {
+  grouping frr-nexthop-attributes {
     leaf nh-type {
       type nexthop-type;
       mandatory true;
       description
         "The nexthop type.";
     }
-
+    leaf vrf {
+      type frr-vrf:vrf-ref;
+      description
+        "The nexthop vrf name, if different from the route.";
+    }
     leaf gateway {
-      type gateway-address;
+      when "../nh-type = 'ip4' or ../nh-type = 'ip6'";
+      type frr-nexthop:gateway-address;
       description
         "The nexthop gateway address.";
     }
-
-    leaf vrf {
-      type string {
-        length "1..36";
+    container gateway-interface {
+      when "../nh-type = 'ip4-ifindex' or"
+         + "../nh-type = 'ip6-ifindex'";
+      leaf gateway {
+        type frr-nexthop:gateway-address;
+        description
+          "The nexthop gateway address.";
+      }
+      leaf interface {
+        type frr-interface:interface-ref;
+        description
+          "The nexthop egress interface.";
+      }
+      leaf onlink {
+        type boolean;
+        default false;
+        description
+          "Nexthop is directly connected.";
       }
-      description
-        "The nexthop vrf name, if different from the route.";
     }
 
     leaf interface {
+      when "../nh-type = 'ifindex'";
       type frr-interface:interface-ref;
       description
         "The nexthop egress interface.";
     }
-
     leaf bh-type {
+      when "../nh-type = 'blackhole'";
       type blackhole-type;
       description
         "A blackhole sub-type, if the nexthop is a blackhole type.";
     }
-
     leaf flags {
       type uint32;
       description
         "The nexthop's raw flags value.";
     }
-
-    leaf is-duplicate {
-      type empty;
+    leaf duplicate {
+      type boolean;
+      default false;
       description
         "Duplicate nexthop; will be ignored.";
     }
-
-    leaf is-recursive {
-      type empty;
+    leaf recursive {
```

Comment:
Ditto this; recursiveness is determined by the route resolver (zebra), no point in configuring it